frame/benchmarking/src/lib.rs

```diff
@@ -736,8 +801,14 @@ macro_rules! impl_benchmark {
 					_ => return Err("Could not find extrinsic."),
 				};
 
-				// Add whitelist to DB
-				$crate::benchmarking::set_whitelist(whitelist.to_vec());
+				// Add whitelist to DB including whitelisted caller
+				let mut whitelist = whitelist.to_vec();
+				let whitelisted_caller_key =
+					<frame_system::Account::<T> as frame_support::storage::StorageMap<_,_>>::hashed_key_for(
+						$crate::whitelisted_caller::<T::AccountId>()
+					);
+				whitelist.push(whitelisted_caller_key.into());
+				$crate::benchmarking::set_whitelist(whitelist);
```

Comment:
Ideally, would like to detect the origin of the extrinsic, then if it is signed, whitelist the caller automatically... but I tried and it did not seem doable within rust macros.

Reply:
instead, we just pick a single account, and this is our whitelisted account to use (unless someone manually whitelists)